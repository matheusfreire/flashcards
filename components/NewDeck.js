import React from 'react'
import { View, TouchableOpacity, Text, StyleSheet, Platform } from 'react-native'
import { TextField } from 'react-native-material-textfield'
import { StackNavigator, NavigationActions } from 'react-navigation'
import { connect } from 'react-redux'
import { black, white } from '../utils/colors'

import { addDeck } from '../actions'
import { addNewDeck } from '../utils/api'

function SubmitBtn({ onPress }) {
    return (
        <TouchableOpacity
            style={Platform.OS === 'ios' ? styles.iosSubmitBtn : styles.AndroidSubmitBtn}
            onPress={onPress}>
            <Text style={styles.submitBtnText}>Submit</Text>
        </TouchableOpacity>
    )
}

class NewDeck extends React.Component {
    state = {
        deck: ''
    }
    submit = () => {
        const {deck} = this.state
        this.props.dispatch(addDeck(deck))
        addNewDeck({deck})
        this.listDecks()
    }

    listDecks = () => {
        this.props.navigation.dispatch(NavigationActions.back({
            key: 'NewDeck'
        }))
    }

    render() {
        const {deck} = this.state
        return (
            <View style={styles.container}>
                <Text style={styles.questions}>
                    What is the title of your new deck?
                </Text>
                <View>
                    <TextField label='Title' value={deck} onChangeText={ (deck) => this.setState({ deck }) }/>
                </View>
                <SubmitBtn onPress={this.submit} />
            </View>
        );
    }
}


const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 20,
        backgroundColor: white
    },
    questions: {
        fontSize: 40,
        alignSelf: 'center',
        justifyContent: 'center',
        alignItems: 'center',
        textAlign: 'center',
    },
    iosSubmitBtn: {
        backgroundColor: black,
        padding: 10,
        borderRadius: 7,
        height: 45,
        marginLeft: 40,
        marginRight: 40,
    },
    AndroidSubmitBtn: {
        backgroundColor: black,
        padding: 10,
        paddingLeft: 30,
        paddingRight: 30,
        height: 45,
        borderRadius: 7,
        alignSelf: 'center',
        justifyContent: 'center',
        alignItems: 'center',
    },
    submitBtnText: {
        color: white,
        fontSize: 22,
        textAlign: 'center',
    },
    center: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        marginLeft: 30,
        marginRight: 30,
    },
    snackbar: {
        flex: 1,
        justifyContent: 'flex-end',
        alignItems: 'flex-end'
    }
})
function mapStateToProps(state) {
    const {deck} =  state
    return {
        deck
    }
}

export default connect(
    mapStateToProps
)(NewDeck)