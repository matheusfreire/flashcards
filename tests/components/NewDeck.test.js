import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import MockAsyncStorage from 'mock-async-storage'
import { AsyncStorage as storage } from 'react-native'

import NewDeck from '../../components/NewDeck';

const middlewares = [thunk]
const mockStore = configureStore(middlewares)
const store = mockStore({})
const props = {}

const mock = () => {
    const mockImpl = new MockAsyncStorage()
    jest.mock('AsyncStorage', () => mockImpl)
}


describe('New deck', () => {

    it('renders without crashing', async () => {
        const rendered = renderer.create(<NewDeck {...props} store={store} />).toJSON();
        expect(rendered).toMatchSnapshot();
    });

})