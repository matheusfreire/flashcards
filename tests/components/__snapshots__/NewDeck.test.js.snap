// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New deck renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "fontSize": 40,
        "justifyContent": "center",
        "textAlign": "center",
      }
    }
  >
    What is the title of your new deck?
  </Text>
  <View>
    <View
      onResponderRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
      style={undefined}
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "rgba(0, 0, 0, 0.38)",
            "borderBottomWidth": 0.5,
            "height": 64,
            "paddingBottom": 8,
            "paddingTop": 32,
          }
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "position": "absolute",
              "top": 36,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.38)",
                "fontSize": 16,
              }
            }
          >
            Title
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="sentences"
            disableFullscreenUI={true}
            editable={true}
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onContentSizeChange={[Function]}
            onFocus={[Function]}
            selectionColor="rgb(0, 145, 234)"
            style={
              Array [
                Object {
                  "flex": 1,
                  "margin": 0,
                  "padding": 0,
                  "top": 2,
                },
                Object {
                  "color": "rgba(0, 0, 0, .87)",
                  "fontSize": 16,
                  "height": 24,
                  "textAlign": "left",
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        collapsable={undefined}
        style={
          Object {
            "flexDirection": "row",
            "height": 8,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "bottom": 0,
                "left": 0,
                "paddingVertical": 4,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "rgb(213, 0, 0)",
                  "fontSize": 0,
                  "opacity": 0,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "bottom": 0,
                "left": 0,
                "paddingVertical": 4,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "rgba(0, 0, 0, .38)",
                  "fontSize": 12,
                  "opacity": 1,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000000",
        "borderRadius": 7,
        "height": 45,
        "marginLeft": 40,
        "marginRight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 22,
          "textAlign": "center",
        }
      }
    >
      Submit
    </Text>
  </View>
</View>
`;
